### PR TITLE
Additional crate documentation about usage fundamentals.

### PR DESCRIPTION
**Pull request by Lokathor**
This adds a small amount of extra documentation to the crate level docs about how to get started using the crate.

### CODE REVIEW

File: logos/src/lib.rs [hunk 1]
```diff
@@ -19,8 +19,22 @@
...
+//!
+//! * One of the `enum` variants must be marked as the `#[error]` variant.
+//!   This is the token that is returned when lexing fails.
+//! * Mark a variant as a `#[regex("string")]` to match via a regular expression.
+//! * Mark a variant as a `#[token("string")]` to match any literal string
+//!   without having to worry about escaping special regex characters such as `+` and `*`.
+//! * If the token variant holds data (eg, `Punctuation(char)`) you assign the data using a callback.
+//!   Put a `,` after your token or regex string,
+//!   and then either name a callback function (eg: `logos::skip`),
+//!   or write a callback expression inline (`eg: `|lex| lex.slice().chars().next().unwrap()`).
```
Comment (domenicquirl) [inline logos/src/lib.rs:33]:
```suggestion
//!   or write a callback expression inline (eg: `|lex| lex.slice().chars().next().unwrap()`).
```
Apologies for the unplanned drive-by review.

File: logos/src/lib.rs [hunk 2]
```diff
@@ -19,8 +19,22 @@
...
+//!   without having to worry about escaping special regex characters such as `+` and `*`.
+//! * If the token variant holds data (eg, `Punctuation(char)`) you assign the data using a callback.
+//!   Put a `,` after your token or regex string,
+//!   and then either name a callback function (eg: `logos::skip`),
+//!   or write a callback expression inline (eg: `|lex| lex.slice().chars().next().unwrap()`).
+//!   There's more about callbacks below.
+//!
 //! ```rust
-//! use logos::Logos;
+//! use logos::{Logos, Lexer};
```
Comment (maciejhirsz) [inline logos/src/lib.rs:37]:
```suggestion
//! use logos::Logos;
```

File: logos/src/lib.rs [hunk 3]
```diff
@@ -45,7 +59,7 @@
 //! }
 //!
 //! fn main() {
-//!     let mut lex = Token::lexer("Create ridiculously fast Lexers.");
+//!     let mut lex: Lexer<_> = Token::lexer("Create ridiculously fast Lexers.");
```
Comment (maciejhirsz) [inline logos/src/lib.rs:62]:
```suggestion
//!     let mut lex = Token::lexer("Create ridiculously fast Lexers.");
```

File: logos/src/lib.rs (hunk 2 above)
Comment (maciejhirsz) [inline logos/src/lib.rs:37]:
```suggestion
//! use logos::Logos;
```

**Review by maciejhirsz — CHANGES_REQUESTED** [review]
I get that these make it easier to understand what's going on, but it might make users think that's what they _have to do_, which would be misleading. Explaining the type in a comment would be better IMHO.

**Pull request closed without merging**